File: src/data/courses/dsa/graphs/shortest-path/cheapest-flights/code/python/cheapest_flights.py
# ============================================================
# Cheapest Flights Within K Stops
# ============================================================
class Solution:
    def findCheapestPrice(self, n, flights, src, dst, k) -> int:
        dist = [float('inf')] * n
        dist[src] = 0
        # Bellman-Ford: k+1 iterations
        for _ in range(k + 1):
            prev = dist[:]  # Use previous round's distances
            for u, v, w in flights:
                if prev[u] + w < dist[v]:
                    dist[v] = prev[u] + w
        return dist[dst] if dist[dst] < float('inf') else -1

File: src/data/courses/dsa/graphs/shortest-path/cheapest-flights/code/java/CheapestFlights.java
// ============================================================
// Cheapest Flights Within K Stops
// ============================================================
import java.util.*;
class Solution {
    public int findCheapestPrice(int n, int[][] flights, int src, int dst, int k) {
        int[] dist = new int[n]; Arrays.fill(dist, Integer.MAX_VALUE); dist[src] = 0;
        for (int i = 0; i <= k; i++) {
            int[] prev = dist.clone();
            for (int[] f : flights) {
                if (prev[f[0]] != Integer.MAX_VALUE && prev[f[0]]+f[2] < dist[f[1]])
                    dist[f[1]] = prev[f[0]]+f[2];
            }
        }
        return dist[dst] == Integer.MAX_VALUE ? -1 : dist[dst];
    }
}
